node/tests/ServerModules.test.ts

```diff
@@ -1161,5 +1164,219 @@ describe("Server Module Tests", () => {
+                                key: Buffer.from("vec"),
+                                value: binaryValue1,
+                            },
+                            {
+                                key: Buffer.from("__VEC_score"),
+                                value: Buffer.from("0"),
+                            },
+                        ],
+                    },
+                ],
+            ];
+            expect(binaryResult).toEqual(expectedBinaryResult);
+        });
+
+        it("FT.SEARCH string test", async () => {
+            client = await GlideClusterClient.createClient(
+                getClientConfigurationOption(
+                    cluster.getAddresses(),
+                    ProtocolVersion.RESP3,
+                ),
+            );
+
+            const prefix = "{" + uuidv4() + "}:";
+            const index = prefix + "index";
+
+            // set string values
+            expect(
+                await GlideJson.set(
+                    client,
+                    prefix + "1",
+                    "$",
+                    '[{"arr": 42}, {"val": "hello"}, {"val": "world"}]',
+                ),
+            ).toEqual("OK");
+
+            // setup a json index:
+            expect(
+                await GlideFt.create(
+                    client,
+                    index,
+                    [
+                        {
+                            type: "NUMERIC",
+                            name: "$..arr",
+                            alias: "arr",
+                        },
+                        {
+                            type: "TEXT",
+                            name: "$..val",
+                            alias: "val",
+                        },
+                    ],
+                    {
+                        dataType: "JSON",
+                        prefixes: [prefix],
+                    },
+                ),
+            ).toEqual("OK");
+
+            // let server digest the data and update index
+            const sleep = new Promise((resolve) =>
+                setTimeout(resolve, DATA_PROCESSING_TIMEOUT),
+            );
+            await sleep;
+
+            const stringResult: (
+                | number
+                | GlideRecord<GlideString | GlideRecord<GlideString>>
+            )[] = await GlideFt.search(client, index, "*", {
+                returnFields: [
+                    { fieldIdentifier: "arr", alias: "myarr" },
+                    { fieldIdentifier: "val", alias: "myval" },
+                ],
+                timeout: 10000,
+                decoder: Decoder.String,
+                limit: { offset: 0, count: 2 },
+            });
+            const expectedStringResult: (
+                | number
+                | GlideRecord<GlideString | GlideRecord<GlideString>>
+            )[] = [
+                1,
+                [
+                    {
+                        key: prefix + "1",
+                        value: [
```

Comment:
Test is failing in github actions: https://github.com/valkey-io/valkey-glide/actions/runs/11602557505/job/32307814737?pr=2551